### PR TITLE
doc: Harmonize guidance on git, cloning, and other workflows.

### PR DESCRIPTION
**Pull request by gwhitney**
By submitting this PR, I am indicating to the Numberscope maintainers that I have read and understood the [contributing guidelines](https://numberscope.colorado.edu/doc/CONTRIBUTING/) and that this PR follows those guidelines to the best of my knowledge. I have also read the [pull request checklist](https://numberscope.colorado.edu/doc/doc/pull-request-checklist/) and followed the instructions therein.

<hr/>

Resolves https://github.com/numberscope/frontscope/issues/221.

There is still a signifcant amount of repetition/overlap among
CONTRIBUTING.md, and in the doc/ directory, onboarding.md,
running-from-source.md, and working-with-git-and-github.md. I wasn't
sure how we might compartmentalize these, because they are at
different levels of detail and for different audiences.

Since they are all now consistent with each other, I am not too fussed
by the repetitiveness. Maybe it's OK to give potential contributors multiple
places to find the information they need. But if the reviewer has suggestions
about how to reduce the redundancy, I am also open to making more changes.

Note this is a strictly doc-only PR. No operational source code is touched.

### CODE REVIEW

File: doc/working-with-git-and-github.md [hunk 1]
```diff
@@ -1,91 +1,342 @@
...
+Note that we have used a different branch in this example because you should
+_never make changes directly to main_. Always create and work in a different
+branch.
+
+### Create a branch
+
+To create a branch and switch to it in one command, first make sure (for
+example, using `git status`) that you are "on" the branch that you want your
+new branch based on. That's sort of the "parent" of the new branch -- in other
+words, the baseline that your branch will represent changes to. Typically for
```
Comment (katestange) [inline doc/working-with-git-and-github.md:157]: what does the word "represent" mean here?  maybe rephrase...?

File: doc/working-with-git-and-github.md [hunk 2]
```diff
@@ -1,91 +1,342 @@
...
+the commit will complete. In these descriptions, always describe the commit in
+present tense: "This commit implements a frobozzinator", not "This commit
+implemented a frobozzinator". Having them in a uniform format makes the
+eventual git log of changes much easier to read.
+
+Note that if you are working on frontscope and you have installed the standard
+pre-commit actions (as will happen automatically if you execute the usual
+`npm install` when you start working with your clone), then before the commit
+actually occurs, git will run several
+[code integrity checks](./husky-pre-commit.md).
```
Comment (katestange) [inline doc/working-with-git-and-github.md:231]: reference to some info on what to do if these fail?

File: doc/working-with-git-and-github.md [hunk 3]
```diff
@@ -1,91 +1,342 @@
...
+left. Then in the bar just below that dropdown list, you should see
+information on how your branch compares with the current "main" branch of
+Numberscope, in terms of the number of commits it is "ahead" of main (i.e.,
+has added since the last common commit to main) and the number of commits
+"behind" main (i.e., the number of commits added in main since the last common
+commit). Ideally, your branch is not behind main at all; otherwise, maybe some
+of those commits in main might affect the changes you were working on. So if
+you see any commits behind main, you might want to go back to your working
+copy and [sync your clone with the original](#sync-a-local-clone) and then
+rebase your branch on the current version of main.
```
Comment (katestange) [inline doc/working-with-git-and-github.md:339]: what happens if you don't go back?

File: doc/working-with-git-and-github.md [hunk 4]
```diff
@@ -153,131 +411,181 @@ To unstash your changes, issue the following command:
...
-original repository.
+And if you previously stashed any changes, you may want to
+[unstash them](#unstash-your-changes).
+
+### Rebase your branch
 
-### Sync local fork with remote original
+A branch works from a specific commit in its parent branch (usually `main`) --
+its like a logbook of all of the changes you've made from that particular
+snapshot of all of the code. Many times for a PR to succeed, the branch must
```
Comment (katestange) [inline doc/working-with-git-and-github.md:498]: Often, for a PR to succeed.... (I had to read this sentence twice for some reason with the current wording)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -15,26 +15,28 @@ Numberscope.
 ## If you are experienced, follow these steps...
```
Comment (katestange) [inline CONTRIBUTING.md:15]: the video playlist doesn't exist

**Review by katestange — CHANGES_REQUESTED** [review]
BTW, might take this opportunity to fix the p5.js link on the building a visualizer markdown doc.

File: CONTRIBUTING.md (hunk 5 above)
Comment (gwhitney) [reply CONTRIBUTING.md:15]: Oh, I had no idea. I've never bothered to click on the link, since video tutorials are basically never of interest to me... I will just remove that section.

**Comment by gwhitney** [comment]
OK, I have attempted to make changes responsive to all of the comments above. Ready for further review (I hope ;-).

**Review by katestange — APPROVED** [review]

**Comment by katestange** [comment]
Looks great, Glen.  I actually learned a lot reading through all that.  I'm not worried about the repetitiveness right now.
